scripts/add_electricity.py

```diff
@@ -327,100 +337,99 @@ def attach_hydro(n, costs, ppl):
 
     _add_missing_carriers_from_costs(n, costs, carriers)
 
-    ppl = ppl.loc[ppl['Fueltype'] == 'Hydro']
-    ppl = ppl.set_index(pd.RangeIndex(len(ppl)).astype(str) + ' hydro', drop=False)
-
-    ppl = ppl.rename(columns={'Capacity':'p_nom', 'Technology': 'technology'})
-    ppl = ppl.loc[ppl.technology.notnull(), ['bus', 'p_nom', 'technology']]
-
-    ppl = ppl.assign(
-        has_inflow=ppl.technology.str.contains('Reservoir|Run-Of-River|Natural Inflow'),
-        has_store=ppl.technology.str.contains('Reservoir|Pumped Storage'),
-        has_pump=ppl.technology.str.contains('Pumped Storage')
-    )
+    ppl = ppl.query('carrier == "hydro"').reset_index(drop=True)\
+             .rename(index=lambda s: str(s) + ' hydro')
+    ror = ppl.query('technology == "Run-Of-River"')
+    phs = ppl.query('technology == "Pumped Storage"')
+    hydro = ppl.query('technology == "Reservoir"')
 
     country = ppl['bus'].map(n.buses.country).rename("country")
 
-    if ppl.has_inflow.any():
-        dist_key = ppl.loc[ppl.has_inflow, 'p_nom'].groupby(country).transform(normed)
+    inflow_idx = ror.index ^ hydro.index
```

Comment:
```suggestion
    inflow_idx = ror.index | hydro.index
```
`^` seems to be an `xor`, ie include it if it in exactly one of the two indices. in principle here one would want rather a full `or`, ie if its in one of the two. in this special case they are equivalent since their "schnitt" (`&`) is empty.

Reply:
ah yes, thanks for clarifying the difference.